### PR TITLE
percona: add some tools and new versions

### PR DESCRIPTION
**Pull request by mkg20001**
###### Motivation for this change

Add some tools and new versions of percona SQL server

// cc @dpausp 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/sql/percona/5.6.x.nix [hunk 1]
```diff
@@ -1,27 +1,28 @@
-{ lib, stdenv, fetchurl, cmake, bison, ncurses, openssl, zlib, libaio, perl }:
+{ lib, stdenv, fetchurl, cmake, boost, bison, ncurses, openssl, readline, zlib, perl }:
+
+# Note: zlib is not required; MySQL can use an internal zlib.
```
Comment (SuperSandro2000) [inline pkgs/servers/sql/percona/5.6.x.nix:3]: Does this mean we can remove it from buildInputs?

File: pkgs/servers/sql/percona/5.6.x.nix [hunk 2]
```diff
@@ -38,11 +39,17 @@ stdenv.mkDerivation rec {
   ];
 
   NIX_CFLAGS_COMPILE = [ "-Wno-error=address-of-packed-member" ];
-  NIX_LDFLAGS = "-lgcc_s";
+  NIX_LDFLAGS = lib.optionalString stdenv.isLinux "-lgcc_s";
 
   prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/servers/sql/percona/5.6.x.nix:44]:
```suggestion
  postPatch = ''
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -21041,6 +21041,8 @@ with pkgs;
   nginx-sso = callPackage ../servers/nginx-sso { };
 
   percona-server56 = callPackage ../servers/sql/percona/5.6.x.nix { };
+  percona-server57 = callPackage ../servers/sql/percona/5.7.x.nix { };
+  percona-server80 = callPackage ../servers/sql/percona/8.0.x.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:21045]:
```suggestion
  percona-server56 = callPackage ../servers/sql/percona/5.6.nix { };
  percona-server57 = callPackage ../servers/sql/percona/5.7.nix { };
  percona-server80 = callPackage ../servers/sql/percona/8.0.nix { };
```

File: pkgs/tools/admin/innotop/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  patches = [ ./innotop.patch ];
+
+  outputs = [ "out" ];
+
+  # The script uses usr/bin/env perl and the Perl builder adds PERL5LIB to it.
+  # This doesn't work. Looks like a bug in Nixpkgs.
+  # Replacing the interpreter path before the Perl builder touches it fixes this.
+  postPatch = ''
+    patchShebangs .
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/innotop/default.nix:26]: Which files do need patching or which directory of files?

File: pkgs/tools/admin/innotop/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  '';
+
+  propagatedBuildInputs = with perlPackages; [ DBI DBDmysql TermReadKey ];
+
+  meta = with lib; {
+    homepage = "https://github.com/innotop/innotop";
+    description = "'top' clone for MySQL with many features and flexibility";
+    license = licenses.gpl2;
+    maintainers = with maintainers; [ ];
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/innotop/default.nix:35]: Please add yourself to maintainers.

File: pkgs/servers/sql/percona/8.0.nix [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  '';
+
+  passthru.mysqlVersion = "8.0";
+
+  meta = with lib; {
+    homepage = "https://www.percona.com";
+    description = "a free, fully compatible, enhanced, open source drop-in replacement for MySQL that provides superior performance, scalability and instrumentation";
+    platforms = platforms.linux;
+    license = licenses.gpl2;
+    maintainers = with maintainers; [ ];
```
Comment (SuperSandro2000) [inline pkgs/servers/sql/percona/8.0.nix:95]: Please add yourself to maintainers.

File: pkgs/servers/sql/percona/8.0.nix [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    "-DWITH_ARCHIVE_STORAGE_ENGINE=1"
+    "-DWITH_BLACKHOLE_STORAGE_ENGINE=1"
+    "-DWITH_INNOBASE_STORAGE_ENGINE=1"
+    "-DWITHOUT_EXAMPLE_STORAGE_ENGINE=1"
+  ];
+
+  NIX_LDFLAGS = lib.optionalString stdenv.isLinux "-lgcc_s";
+  CXXFLAGS = lib.optionalString stdenv.isi686 "-fpermissive";
+
+  prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/servers/sql/percona/8.0.nix:68]:
```suggestion
  postPatch = ''
```

File: pkgs/servers/sql/percona/5.7.nix [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    "-DINSTALL_DOCREADMEDIR=share/mysql"
+    "-DINSTALL_SUPPORTFILESDIR=share/mysql"
+    "-DINSTALL_MYSQLSHAREDIR=share/mysql"
+    "-DINSTALL_DOCDIR=share/mysql/docs"
+    "-DINSTALL_SHAREDIR=share/mysql"
+  ];
+
+  NIX_LDFLAGS = lib.optionalString stdenv.isLinux "-lgcc_s";
+
+  prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/servers/sql/percona/5.7.nix:54]:
```suggestion
  postPatch = ''
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Can we use a common nix file instead of copying?

**Comment by ajs124** [comment]
Why was this closed?

**Pull request closed without merging**